src/Components/Aspire.Microsoft.Azure.Cosmos/AspireMicrosoftAzureCosmosExtensions.cs

```diff
@@ -119,10 +197,46 @@ private static void AddAzureCosmosClient(
         if (serviceKey is null)
         {
             builder.Services.AddSingleton(_ => ConfigureDb());
+            // Configure database and container services if specified
+            if (!string.IsNullOrEmpty(databaseName) && string.IsNullOrEmpty(containerName))
+            {
+                builder.Services.AddSingleton(sp =>
+                {
+                    var client = sp.GetRequiredService<CosmosClient>();
+                    return client.GetDatabase(databaseName);
+                });
+
+            }
```

Comment:
Should the `Database` be added when both it and the container are specified?

Reply:
I think this ties a bit into the conversation we were having about what services get configured in the ServiceBus scenario. 

IMO, it makes sense to limit it to just Client + Container for this overload. You can always resolve the database yourself with the client that we inject.